sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -323,7 +323,7 @@ abstract class StreamExecution(
       startLatch.countDown()
 
       // While active, repeatedly attempt to run batches.
-      SparkSession.setActiveSession(sparkSession)
+      SparkSession.setActiveSession(sparkSessionForStream)
```

Comment:
Can we have a test case by using your PR description, @Ngone51 ?

Reply:
ok, let me try it.